circuit/src/execution-delivery/src/exec_composer/mod.rs

```diff
@@ -849,4 +892,303 @@ mod tests {
+            assert_eq!(unwrapped_result.0.len(), 0);
+        });
+    }
+
+    #[test]
+    fn preload_bunch_of_contracts_one_contract_succeeds() {
+        let mut ext = TestExternalities::new_empty();
+        let requester =
+            AccountId32::from_str("5G9VdMwXvzza9pS8qE8ZHJk3CheHW9uucBn9ngW4C1gmmzpv").unwrap();
+        let value = BalanceOf::<Test>::from(0u32);
+        let temp_contract_one =
+            make_registry_contract_out_of_wat::<Test>(CODE_CALL, vec![], requester.clone(), value);
+
+        let schedule = <Test as VolatileVM>::Schedule::get();
+        let executable = PrefabWasmModule::<Test>::from_code(
+            temp_contract_one.bytes.clone(),
+            &schedule,
+            RunMode::Dry,
+            None,
+        )
+        .unwrap();
+
+        ext.execute_with(|| {
+            insert_default_xdns_record();
+            let res = ExecComposer::preload_bunch_of_contracts::<Test>(
+                vec![temp_contract_one],
+                Default::default(),
+            );
+
+            assert_ok!(res);
+            // If we are able to unwrap, it means the contract was inserted.
+            let _fetched_contract =
+                volatile_vm::Pallet::<Test>::get_contract_code_lazy(executable.code_hash).unwrap();
+        });
+    }
+
+    #[test]
+    fn preload_bunch_of_contracts_multiple_contract_succeeds() {
+        let mut ext = TestExternalities::new_empty();
+        let requester =
+            AccountId32::from_str("5G9VdMwXvzza9pS8qE8ZHJk3CheHW9uucBn9ngW4C1gmmzpv").unwrap();
+        let value = BalanceOf::<Test>::from(0u32);
+        let temp_contract_one =
+            make_registry_contract_out_of_wat::<Test>(CODE_CALL, vec![], requester.clone(), value);
+        let temp_contract_two =
+            make_registry_contract_out_of_wat::<Test>(CODE_CALL, vec![], requester.clone(), value);
+        ext.execute_with(|| {
+            insert_default_xdns_record();
+
+            let res = ExecComposer::preload_bunch_of_contracts::<Test>(
+                vec![temp_contract_one, temp_contract_two],
+                Default::default(),
+            );
+
+            assert_ok!(res);
+        });
+    }
+
+    #[test]
+    fn preload_bunch_of_contracts_one_contract_fails() {
+        let mut ext = TestExternalities::new_empty();
+        let requester =
+            AccountId32::from_str("5G9VdMwXvzza9pS8qE8ZHJk3CheHW9uucBn9ngW4C1gmmzpv").unwrap();
+        let value = BalanceOf::<Test>::from(0u32);
+        let temp_contract_one = make_registry_contract_out_of_wat::<Test>(
+            INVALID_CODE,
+            vec![],
+            requester.clone(),
+            value,
+        );
+
+        ext.execute_with(|| {
+            insert_default_xdns_record();
+
+            let res = ExecComposer::preload_bunch_of_contracts::<Test>(
+                vec![temp_contract_one],
+                Default::default(),
+            );
+
+            assert_eq!(res, Err("Can't decode WASM code"));
```

Comment:
You can use assert_err! macro when asserting errors

Reply:
`#[macro_export]
macro_rules! assert_err {
	( $x:expr , $y:expr $(,)? ) => {
		assert_eq!($x, Err($y.into()));
	}
}`

The macro does the same thing. But makes sense, it is always good to be explicit in what we want.
Replaced all the places where Error was being asserted.